service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/spring/KeycloakServiceImpl.java

```diff
@@ -204,6 +207,11 @@ private UserRepresentation create(String userName, String password, boolean enab
 
 			user.setCredentials(Arrays.asList(getUserCredentials(password, false)));
 		}
+        if (userId == null) {
+            logger.warn("create(userName={}) no userId received", userName);
+        } else {
+            user.singleAttribute(USER_ID_PROPERTY, userId.toString());
```

Comment:
it seems that what we have discussed, was not implemented as expected.  
the keycloak's UserRepresintation should have been populated with a list tupels, each containing 
1. `application name` - to serve as key by which to pull values according to sub-domain from which the user contacts the service
2. `applicationId` - the tenant id in essence 
3. `user_id` - as is now implemented